src/jit/lclvars.cpp

```diff
@@ -1434,13 +1448,15 @@ void   Compiler::lvaCanPromoteStructType(CORINFO_CLASS_HANDLE     typeHnd,
             return;
         }
 
-#ifdef _TARGET_ARM_        
-        // For ARM don't struct promote if we have an CUSTOMLAYOUT flag on an HFA type 
+#ifdef FEATURE_HFA
```

Comment:
This #ifdef is not required if IsHfa(hClass) is defined to return false for non-hfa targets.